lib/trento/application/integration/discovery/protocol/enrich_request_host_deregistration.ex

```diff
@@ -17,32 +22,41 @@ defimpl Trento.Support.Middleware.Enrichable,
           | {:error, :host_alive}
           | {:error, :host_not_registered}
   def enrich(%RequestHostDeregistration{host_id: host_id} = command, _) do
-    Repo.get(HostReadModel, host_id)
-    |> Repo.preload([:heartbeat_timestamp])
+    HostReadModel
+    |> where([h], h.id == ^host_id)
+    |> enrich_host_read_model_query()
+    |> Repo.one()
     |> host_deregisterable(command)
   end
 
   defp host_deregisterable(
-         %HostReadModel{heartbeat_timestamp: nil, deregistered_at: nil},
+         %HostReadModel{last_heartbeat_timestamp: nil, deregistered_at: nil},
          %RequestHostDeregistration{} = command
        ),
        do: {:ok, command}
 
   defp host_deregisterable(
          %HostReadModel{
-           heartbeat_timestamp: %Trento.Heartbeat{timestamp: timestamp},
+           last_heartbeat_timestamp: last_heartbeat_timestamp,
            deregistered_at: nil
          },
          %RequestHostDeregistration{} = command
        ) do
     if :lt ==
          DateTime.compare(
            DateTime.utc_now(),
-           DateTime.add(timestamp, @total_deregistration_debounce, :millisecond)
+           DateTime.add(last_heartbeat_timestamp, @total_deregistration_debounce, :millisecond)
          ),
        do: {:error, :host_alive},
        else: {:ok, command}
   end
 
   defp host_deregisterable(_, _), do: {:error, :host_not_registered}
+
+  @spec enrich_host_read_model_query(Ecto.Query.t()) :: Ecto.Query.t()
+  defp enrich_host_read_model_query(query) do
+    query
+    |> join(:left, [h], hb in Heartbeat, on: type(h.id, :string) == hb.agent_id)
```

Comment:
As discussed: since we reuse this helper function in both `enrich_host_read_model_query()` in `get_all_hosts()` and `get_host_by_id()`, using a `JOIN` like this offers a more flexible and reusable solution.